chasm/lib/activity/activity.go

```diff
@@ -16,8 +18,11 @@ import (
 )
 
 type ActivityStore interface {
+	// PopulateRecordActivityTaskStartedResponse populates the response for RecordActivityTaskStarted
 	PopulateRecordActivityTaskStartedResponse(ctx chasm.Context, key chasm.EntityKey, response *historyservice.RecordActivityTaskStartedResponse) error
-	RecordCompletion(ctx chasm.MutableContext) error
+
+	// RecordCompletion applies the provided function to record activity completion
+	RecordCompletion(ctx chasm.MutableContext, applyFn func(ctx chasm.MutableContext) error) error
```

Comment:
This is likely to change when we have an embedded-within-workflow implementation but don't stress about that too much now.